File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

string encrypt(int key, string plaintext)
{

    for (int i = 0, length = strlen(plaintext); i < length; i++)
    {

        // Check that the char is a lower letter
        if (plaintext[i] >= 'a' && plaintext[i] <= 'z')
        {
            // Cipher in the space of lowers letter
            plaintext[i] = 'a' + (plaintext[i] - 'a' + key) % ('a' - 'z' - 1);
        }

        // Check that the char is a upper letter
        if (plaintext[i] >= 'A' && plaintext[i] <= 'Z')
        {
            // Ciphe in the space of upper letter
            plaintext[i] = 'A' + (plaintext[i] + key - 'A') % ('A' - 'Z' - 1);
        }
    }

    return plaintext;
}

int main(int argc, string argv[])
{
    // Check if there is only one argument
    if (argc != 2)
    {
        printf("Usage ./%s key\n", argv[0]);
        return 1;
    }

    // Check that the key is a number
    for (int i = 0, length = strlen(argv[1]); i < length ; i++)
    {
        if (argv[1][i] < '0' || argv[1][i] > '9')
        {
            printf("Usage ./%s key\n", argv[0]);
            return 1;
        }
    }

    // Assing the key
    int key = atoi(argv[1]);

    // Ask for the text to ciphe
    string plain_text = get_string("plaintext:");

    // Print the ciphertext
    printf("ciphertext: %s\n", encrypt(key, plain_text));

    return 0;
}

File: pset6/caesar/caesar.py
from sys import argv, exit
from cs50 import get_string


def main():
    # Check amount of arguments
    if len(argv) != 2:
        print("usage:")
        print(f"python {argv[0]} k")
        exit(1)
    # Get the key and convert to int
    key = int(argv[1])

    # Prompt user
    plain_text = get_string("plaintext: ")
    # Encrypt and print the cipher text
    print(f"ciphertext: {encrypt(key, plain_text)}")


def encrypt(key, text):
    cipher_text = ""
    # Iterate over every character to check if it's alphabetic
    for c in text:
        if c.isalpha():
            # Set value to translate to the space 0-26 acording if it's lower or upper
            if c.isupper():
                mapper = ord('A')
            elif c.islower():
                mapper = ord('a')
            # Translate to the space 0-26
            c = ord(c) - mapper
            # Cipher
            c = (c + key) % 26
            # Translate to the char space and convert to char
            c = chr(c + mapper)
        cipher_text += c

    return cipher_text


if __name__ == "__main__":
    main()